Minio/Credentials/IAMAWSProvider.cs

```diff
@@ -0,0 +1,265 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage,
+ * (C) 2021 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using RestSharp;
+using System.Linq;
+using System.Net;
+
+using Minio.DataModel;
+using Minio.Exceptions;
+using Newtonsoft.Json;
+using System.IO;
+using Newtonsoft.Json.Serialization;
+using System.Collections.Generic;
+
+/*
+ * IAM roles for Amazon EC2
+ * http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+ * The Credential provider for attaching an IAM rule.
+ */
+
+namespace Minio.Credentials
+{
+    public class IAMAWSProvider : EnvironmentProvider
+    {
+        internal Uri CustomEndPoint { get; set; }
+        internal AccessCredentials Credentials { get; set; }
+        internal MinioClient Minio_Client { get; set; }
+
+        public override AccessCredentials GetCredentials()
+        {
+            this.Validate();
+            Uri url = this.CustomEndPoint;
+            if (this.CustomEndPoint == null)
+            {
+                string region = Environment.GetEnvironmentVariable("AWS_REGION");
+                if (string.IsNullOrEmpty(region) || string.IsNullOrWhiteSpace(region))
```

Comment:
```suggestion
                if ( string.IsNullOrWhiteSpace(region))
```

Reply:
This change has been made everywhere in this PR, wherever `string.IsNullOrEmpty()` had been used.